### DISCUSSION TITLE
Is it possible to run Stylelint against .css.ts files?

### DISCUSSION THREAD

**Discussion by stormwarning** [vanilla-extract-css/vanilla-extract#858] [Q&A]
I imagine some Stylelint rules wouldn't work (or make sense) in a vanilla-extract setup, but I'm not even sure how to get Stylelint to recognise the styles.

Anyone have a setup that's working for them?

**Comment by rc-glean** [comment]
FYI, there's some work to support this https://github.com/stylelint/stylelint/issues/6489#issuecomment-1326659603:

> As far as I'm aware, no one has written a dedicated PostCSS syntax for an object-based CSS-in-JS library yet.

**Reply by osdiab** [reply]
Playing around with what it would take to make a postcss parser/stringifier. Basing off of the [Linaria one](https://github.com/callstack/linaria/tree/master/packages/postcss-linaria/src), they basically just look at all template string literals and use some heuristics to decide if a token inside of one is a selector, expression, etc in order to produce a PostCSS AST.

Seems pretty tricky to do this with Vanilla Extract though given how flexibly you can make styles with full JS expressivity - maybe a good starting point is to look for any objects created with the `style` function in its most basic usage and slowly build it up to accept all the various ways of expressing things? but not really sure how far it's possible to go this route.
